pagefaultgames/pokerogue
# [QoL][DEV] add PokemonMoveset type

## What are the changes?

I've replaced all the `Array<Moves>` types with a new `PokemonMoveset` type which has a tuple value of 
```ts
[Moves?, Moves?, Moves?, Moves?]
```

## Why am I doing these changes?




It informs the dev about using too many array indexes. The compiler will complain if more than 4 moves are used

## What did change?



I've replaced all the `Array<Moves>` types with a new `PokemonMoveset` type/tuple;
```ts
type PokemonMoveset = [Moves?, Moves?, Moves?, Moves?];
```

### Screenshots/Videos



## How to test the changes?




Honestly...
See if tests & build still runs.
I don't see anything else changing as this has no influence of the output js file.

## Checklist
- ~~Are the changes visual?~~
  
  
<img width="313" alt="image" src="https://github.com/pagefaultgames/pokerogue/assets/50131232/f09bfe8c-bb28-4ae0-a094-85458ffe1f10">
<img width="458" alt="image" src="https://github.com/pagefaultgames/pokerogue/assets/50131232/d1ed31f3-18c9-4614-84cd-551c2723765b">
